### DISCUSSION TITLE
Rule-based entity recognition for non-default english models

### DISCUSSION THREAD

**Discussion by transfluxus** [explosion/spaCy#11422] [Help: Coding & Implementations]
## How to reproduce the behaviour

Using the 1. example of entityruler-usage
[Using the entity ruler](https://spacy.io/usage/rule-based-matching#entityruler-usage)

Trying it out with the following 3 models:

```
for model in ["en_core_web_sm", "en_core_web_md", "en_core_web_lg"]:
    print(model)
    nlp = spacy.load(model)
    ruler = nlp.add_pipe("entity_ruler")
    patterns = [{"label": "ORG", "pattern": "Apple"},
                {"label": "GPE", "pattern": [{"LOWER": "san"}, {"LOWER": "francisco"}]}]
    ruler.add_patterns(patterns)

    doc = nlp("Apple is opening its first big office in San Francisco.")
    print([(ent.text, ent.label_) for ent in doc.ents])
```

the results are:

> en_core_web_sm
> [('Apple', 'ORG'), ('first', 'ORDINAL'), ('San Francisco', 'GPE')]
> en_core_web_md
> [('Apple', 'ORG'), ('first', 'ORDINAL'), ('San Francisco', 'GPE')]
> en_core_web_lg
> [('Apple', 'ORG'), ('first', 'ORDINAL'), ('San Francisco', 'GPE')]

Furthermore, when I try to add ids to the patterns like in the next section the `ent_id_` are simple empty strings.

## Info about spaCy

- **spaCy version:** 3.4.1
- **Platform:** Linux-5.13.0-21-generic-x86_64-with-glibc2.34
- **Python version:** 3.9.7
- **Pipelines:** en_core_web_sm (3.4.0), es_core_news_md (3.4.0), en_core_web_md (3.4.0), pt_core_news_sm (3.4.0), pt_core_news_md (3.4.0), es_core_news_sm (3.4.0), es_core_news_lg (3.4.0), en_core_web_lg (3.4.0), pt_core_news_lg (3.4.0), fr_core_news_md (3.4.0)

**Comment by transfluxus** [comment]
Maybe to add:
When I use the default model with 

```
    from spacy.lang.en import English
    nlp = English()
```

the results are as expected

> [('Apple', 'ORG', 'apple'), ('San Francisco', 'GPE', 'san-francisco')]
> [('Apple', 'ORG', 'apple'), ('San Fran', 'GPE', 'san-francisco')]
> ...

**Comment by polm** [comment]
By default, the EntityRuler doesn't overwrite entities. That means that entities detected by the pretrained pipelines will not be modified, which includes not giving them entity IDs. You can use the `overwrite_ents` setting to change this, see [the docs](https://spacy.io/api/entityruler/#config).

Moving to Discussions because this is not a bug report.
